x/stakeibc/keeper/msg_server_submit_tx.go

```diff
@@ -91,6 +94,77 @@ func (k Keeper) DelegateOnHost(ctx sdk.Context, hostZone types.HostZone, amt sdk
 	return nil
 }
 
+func (k Keeper) ReinvestRewards(ctx sdk.Context, hostZone types.HostZone) error {
+	_ = ctx
+	// the relevant ICA is the delegate account
+	owner := types.FormatICAAccountOwner(hostZone.ChainId, types.ICAAccountType_DELEGATION)
+	portID, err := icatypes.NewControllerPortID(owner)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "%s has no associated portId", owner)
+	}
+	connectionId, err := k.GetConnectionId(ctx, portID)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidChainID, "%s has no associated connection", portID)
+	}
+
+	// Fetch the relevant ICA
+	delegationAccount := hostZone.GetDelegationAccount()
+	withdrawAccount := hostZone.GetWithdrawalAccount()
+
+	SubmitTxs := k.SubmitTxs
+	GetParam := k.GetParam
+	cdc := k.cdc
+	
+	var cb icqkeeper.Callback = func(k icqkeeper.Keeper, ctx sdk.Context, args []byte, query icqtypes.Query) error {
+		var msgs []sdk.Msg
+		queryRes := bankTypes.QueryAllBalancesResponse{}
+		err := cdc.Unmarshal(args, &queryRes)
+		if err != nil {
+			k.Logger(ctx).Error("Unable to unmarshal balances info for zone", "err", err)
+			return err
+		}
+		// Get denom dynamically
+		balance := queryRes.Balances.AmountOf(hostZone.HostDenom)
+		balanceDec := sdk.NewDec(balance.Int64())
+		commission := sdk.NewDec(int64(GetParam(ctx, types.KeyStrideCommission)))
+		// Dec type has 18 decimals and the same precision as Coin types
+		strideAmount := balanceDec.Mul(commission)
+		reinvestAmount := balanceDec.Sub(strideAmount)
+		strideCoin := sdk.NewCoin(hostZone.HostDenom, strideAmount.TruncateInt())
+		reinvestCoin := sdk.NewCoin(hostZone.HostDenom, reinvestAmount.TruncateInt())
```

Comment:
Looks like this always rounds down per [here](https://github.com/cosmos/cosmos-sdk/blob/7feae9c4dd0fecceab348ace054561dd0638cf6b/types/decimal.go#L664) so should be safe to truncate balances like this